pallets/asset-manager/src/lib.rs

```diff
@@ -0,0 +1,393 @@
+        fn create_asset(
+            asset: T::AssetId,
+            min_balance: T::Balance,
+            metadata: T::AssetRegistrarMetadata,
+            // Wether or not an asset-receiving account increments the sufficient counter
+            is_sufficient: bool,
+        ) -> DispatchResult;
+    }
+
+    // We implement this trait to be able to get the AssetType and units per second registered
+    impl<T: Config> primitives::xcm_gadget::AssetTypeGetter<T::AssetId, T::AssetType> for Pallet<T> {
+        fn get_asset_type(asset_id: T::AssetId) -> Option<T::AssetType> {
+            AssetIdType::<T>::get(asset_id)
+        }
+
+        fn get_asset_id(asset_type: T::AssetType) -> Option<T::AssetId> {
+            AssetTypeId::<T>::get(asset_type)
+        }
+    }
+
+    impl<T: Config> primitives::xcm_gadget::UnitsToWeightRatio<T::AssetType> for Pallet<T> {
+        fn payment_is_supported(asset_type: T::AssetType) -> bool {
+            SupportedFeePaymentAssets::<T>::get()
+                .binary_search(&asset_type)
+                .is_ok()
+        }
+        fn get_units_per_second(asset_type: T::AssetType) -> Option<u128> {
+            AssetTypeUnitsPerSecond::<T>::get(asset_type)
+        }
+    }
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// The Asset Id. This will be used to register the asset in Assets
+        type AssetId: Member + Parameter + Default + Copy + HasCompact + MaxEncodedLen;
+
+        /// The Asset Metadata we want to store
+        type AssetRegistrarMetadata: Member + Parameter + Default;
+
+        /// The Asset Kind.
+        type AssetType: Parameter + Member + Ord + PartialOrd + Into<Self::AssetId> + Default;
+
+        /// The units in which we record balances.
+        type Balance: Member + Parameter + AtLeast32BitUnsigned + Default + Copy + MaxEncodedLen;
+
+        /// The trait we use to register Assets
+        type AssetRegistrar: AssetRegistrar<Self>;
+
+        /// Origin that is allowed to create and modify asset information
+        type AssetModifierOrigin: EnsureOrigin<Self::Origin>;
+
+        type WeightInfo: WeightInfo;
+    }
+
+    /// An error that can occur while executing the mapping pallet's logic.
+    #[pallet::error]
+    pub enum Error<T> {
+        ErrorCreatingAsset,
+        AssetAlreadyExists,
+        AssetDoesNotExist,
+        TooLowNumAssetsWeightHint,
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(crate) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// New asset with the asset manager is registered
+        AssetRegistered {
+            asset_id: T::AssetId,
+            asset: T::AssetType,
+            metadata: T::AssetRegistrarMetadata,
+        },
+        /// Changed the amount of units we are charging per execution second for a given asset
+        UnitsPerSecondChanged {
```

Comment:
we use `Updated` in other places instead of changed, so I think it can be consistent